pkg/exporter/azureblob_exporter.go

```diff
@@ -19,23 +20,68 @@ const (
 )
 
 // AzureBlobExporter defines an Azure Blob Exporter
-type AzureBlobExporter struct{}
+type AzureBlobExporter struct {
+	BaseExporter
+}
 
 var _ interfaces.Exporter = &AzureBlobExporter{}
 
+// NewAzureBlobExporter is a constructor
+func NewAzureBlobExporter() *AzureBlobExporter {
+	return &AzureBlobExporter{
+		BaseExporter: BaseExporter{
+			exporterType: AzureBlob,
+		},
+	}
+}
+
+// GetStorageContainerName get storage container name
+func (exporter *AzureBlobExporter) GetStorageContainerName(APIServerFQDN string) (string, error) {
+	var containerName string
+	var err error
+	if utils.IsKubernetesInDocker() {
+		containerName, err = exporter.GetKubernetesInDockerStorageContainerName(APIServerFQDN)
+	} else {
+		containerName, err = exporter.GetNonKINDStorageContainerName(APIServerFQDN)
+	}
+
+	//TODO run a sanitizer over the final chars in the containerName
+	return containerName, err
+}
+
+func (exporter *AzureBlobExporter) GetKubernetesInDockerStorageContainerName(APIServerFQDN string) (string, error) {
+	containerName := strings.Replace(APIServerFQDN, ".", "-", -1)
+
+	return containerName, nil
+}
+
+func (exporter *AzureBlobExporter) GetNonKINDStorageContainerName(APIServerFQDN string) (string, error) {
+	containerName := strings.Replace(APIServerFQDN, ".", "-", -1)
+
+	//TODO I really dont like the line below, it makes for weird behaviour if e.g. .hcp. or -hcp- is in the fqdn for some reason other than being auto-added by AKS
+	length := strings.Index(containerName, "-hcp-")
+
+	if length == -1 {
+		maxLength := len(containerName)
+		length = int(math.Min(float64(maxLength), float64(maxContainerNameLength)))
+	}
+
+	containerName = containerName[:length]
+	containerName = strings.TrimRight(containerName, "-")
+	return containerName, nil
+}
+
 // Export implements the interface method
 func (exporter *AzureBlobExporter) Export(files []string) error {
 	APIServerFQDN, err := utils.GetAPIServerFQDN()
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to get APIServerFQDN: %+v", err)
 	}
 
-	containerName := strings.Replace(APIServerFQDN, ".", "-", -1)
-	len := strings.Index(containerName, "-hcp-")
-	if len == -1 {
-		len = maxContainerNameLength
+	containerName, err := exporter.GetStorageContainerName(APIServerFQDN)
+	if err != nil {
+		return fmt.Errorf("Failed to get StorageContainerName: %+v", err)
```

Comment:
`return fmt.Errorf("get StorageContainerName: %w", err)`

Reply:
Actually wont change this now - it is consistent with the rest of the codebase, if we want to change it should all be at once and separately